### ISSUE TITLE
Build failure: emacs-macport hangs during dumping (`temacs`)

### ISSUE THREAD

**Issue by isker** [NixOS/nixpkgs#333731]
### Steps To Reproduce

Run `nix build nixpkgs/aa9cfafc2a8dd21b11359ef32e318246d8de226c#emacs-macport --rebuild --keep-failed --print-build-logs`

(That is the latest commit on `nixpkgs-24.05-darwin`, but it happens on `master` and any other commit I try as well.)

Build hangs during `temacs`:
```
emacs-mac-macport> env -i ./temacs --batch --no-build-details -l loadup --temacs=pbootstrap \
emacs-mac-macport>      --bin-dest /nix/store/bd3j5bkmy0h9aqjnch8cgvk039rh04f1-emacs-mac-macport-29.1/bin/ --eln-dest /nix/store/bd3j5bkmy0h9aqjnch8cgvk039rh04f1-emacs-mac-macport-29.1/lib/emacs/29.1/
```
It just sits there, doing nothing as far as I can tell, indefinitely.

### Build log
https://gist.github.com/isker/fc551686208211b4630db1f242f6cf74

### Additional context
I was trying to build a newer "unreleased" version of the macport from https://bitbucket.org/mituharu/emacs-mac/src/work/, experienced these hangs, and realized they happen on the current version in nixpkgs as well.

I can get past the `temacs` step when building `emacs29` just fine, or when building the macport outside of nix.  (It completes in an instant.)  It's only the macport nix build that seems to suffer from this hang.

I am pretty clueless here, but here's a sample I took from Activity Monitor: https://gist.github.com/isker/6a8c8a1f07a5268c6854dfdf9518df45.

### Notify maintainers

Maintainers that are macport users as far as I'm aware:
@jwiegley (who reports build failures that seemingly come later in the build in #331583, which makes this more interesting)
@Atemu 

If you have any advice for how to debug further, I'd appreciate it.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
λ nix-shell -p nix-info --run "nix-info -m"
error:
       … <borked>

         at «none»:0: (source not available)

       … while calling the 'import' builtin

         at «string»:1:18:

            1| {...}@args: with import <nixpkgs> args; (pkgs.runCommandCC or pkgs.runCommand) "shell" { buildInputs = [ (nix-info) ]; } ""
             |                  ^

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: file 'nixpkgs' was not found in the Nix search path (add it using $NIX_PATH or -I)

       at «none»:0: (source not available)

λ nix shell nixpkgs#nix-info --command nix-info -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - channels(isker): `"darwin, home-manager"`
 - nixpkgs: `not found`
```

But I experience the same hangs on another intel macbook as well.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by Atemu** [comment]
Given that hydra can build it just fine, this must be an issue of your machine.

Any "anti" virus snakeoil or similar garbage daemons running that might interfere?

Other than that, monitor CPU, memory usage and IO while it's running; what's happening on the system?

**Comment by isker** [comment]
No, no antivirus or anything else on the ARM MacBook used in my report. (The intel MacBook I also experience this problem does have the lovely CrowdStrike antivirus on it, but the ARM most certainly does not.)

The system is dead quiet when this happens. As the linked sample gist shows, temacs in particular is doing nothing, it seems to be suspended waiting to receive some kind of message, but I’m pretty ignorant about these things.

Did I choose the wrong issue type? Why have you closed it?

**Comment by Atemu** [comment]
I closed this because the physical mac machines that we use for CI built emacs-macport just fine a few hours before you commented. It must therefore be an issue with your particular system rather than a general issue affecting all macs.

While I'll gladly attempt to help you troubleshoot a bit as there's still potential that there is *some* actual Nixpkgs issue at hand, keep in mind that the Nixpkgs issue tracker's purpose is to keep track of and discuss bugs and feature requests for Nixpkgs, not to provide tech support. The NixOS discourse would be a more appropriate place for this but you likely didn't know that it's a you-problem when creating this issue.

I'm not ignorant to the fact that such issues can be an incredibly helpful resource for third parties when troubleshooting which is why I'm not locking or deleting this but, for the purpose of tracking actual issues, this is just noise, so closing it is the best course of action.  

If other users chime in and report the same issue, you can start figuring out commonalities that do not apply to our builders, figure out the root cause and turn this into an actual issue. Even if it's not our fault (as evidenced by it working in the builders), working around issues of commonly used external software and/or configurations can sometimes be prudent.

**Comment by isker** [comment]
Okay, thanks.

I did find someone else that runs nix with flakes on macOS and they successfully ran the `nix build` command I provided above.  So it must not be terribly hard to get a machine with a correct configuration, I'm just not sure at this point what the two machines I've got have in common.  The fact that this only happens on macport but not the main package is even stranger.

**Comment by isker** [comment]
Good (?) news.  I followed these steps:
1. Uninstalled nix-darwin by running `darwin-uninstaller`.
2. Uninstalled nix according to https://nix.dev/manual/nix/2.22/installation/uninstall#macos.
3. Reinstalled nix using https://github.com/DeterminateSystems/nix-installer.
4. Reinstalled nix-darwin: https://github.com/LnL7/nix-darwin?tab=readme-ov-file#step-2-installing-nix-darwin.

And the build succeeds, not only for 29.1 but also for the not yet officially released 29.4, as I was originally trying to do.

I am not sure what this did other than clearing away old cruft from my nix installation (I did not always use flakes) and upgraded my version of nix beyond `nixVersions.stable`:

```console
λ nix shell nixpkgs#nix-info --command nix-info -m
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.5.0, macOS 14.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.23.3`
 - nixpkgs: `/nix/store/dxfjmv6mq4rndmbhc6fc5bpw92avbjxk-source`
```

Thanks for your help.

**Comment by Atemu** [comment]
Glad you worked it out.
